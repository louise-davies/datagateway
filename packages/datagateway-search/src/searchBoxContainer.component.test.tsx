import React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';

import SearchBoxContainer from './searchBoxContainer.component';
import SearchBoxContainerSide from './searchBoxContainerSide.component';
import { useSelector } from 'react-redux';
import { initialState } from './state/reducers/dgsearch.reducer';

jest.mock('loglevel');

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

describe('SearchBoxContainer - Tests', () => {
  const createWrapper = (): ShallowWrapper => {
    return shallow(
      <SearchBoxContainer
        initiateSearch={jest.fn()}
        onSearchTextChange={jest.fn()}
        searchText=""
      />
    );
  };

  beforeEach(() => {
    useSelector.mockImplementation(() => {
      return initialState;
    });
  });

  it('renders searchBoxContainer correctly', () => {
    const wrapper = createWrapper();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('SearchBoxContainerSide - Tests', () => {
  const createWrapper = (): ShallowWrapper => {
    return shallow(
      <SearchBoxContainerSide
        initiateSearch={jest.fn()}
        onSearchTextChange={jest.fn()}
        searchText=""
      />
    );
  };

  it('renders searchBoxContainerSide correctly', () => {
    const wrapper = createWrapper();
    expect(wrapper).toMatchSnapshot();
  });
});
