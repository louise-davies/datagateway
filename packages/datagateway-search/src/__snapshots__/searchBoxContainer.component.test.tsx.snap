// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBoxContainer - Tests renders searchBoxContainer correctly 1`] = `
<Styled(ForwardRef(Box))>
  <ForwardRef(Grid)
    container={true}
    direction="row"
    id="container-searchbox"
    justifyContent="center"
  >
    <ForwardRef(Grid)
      item={true}
      style={
        Object {
          "flexGrow": 1,
        }
      }
      xs="auto"
    >
      <SearchTextBox
        initiateSearch={[MockFunction]}
        onChange={[MockFunction]}
        searchText=""
      />
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      sx={
        Object {
          "marginTop": "8px",
        }
      }
    >
      <Connect(CheckboxesGroup) />
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      sx={
        Object {
          "marginTop": "8px",
        }
      }
    >
      <Connect(SelectDates)
        initiateSearch={[MockFunction]}
      />
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      sx={
        Object {
          "display": "flex",
          "marginLeft": "6px",
          "marginTop": "24px",
        }
      }
    >
      <TranslatedSearchButton
        initiateSearch={[MockFunction]}
      />
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <ForwardRef(Typography)
      sx={
        Object {
          "color": [Function],
          "fontSize": "14px",
          "textAlign": "left",
        }
      }
    >
      <Trans
        i18nKey="searchBox.examples_label"
        t={[Function]}
      >
        For example
        <ForwardRef(Link)
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          sx={
            Object {
              "fontWeight": "bold",
            }
          }
          to="searchBox.examples_label_link1"
        >
          "instrument calibration"
        </ForwardRef(Link)>
        or
         
        <ForwardRef(Link)
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          sx={
            Object {
              "fontWeight": "bold",
            }
          }
          to="searchBox.examples_label_link2"
        >
          neutron AND scattering
        </ForwardRef(Link)>
        .
      </Trans>
       
      <AdvancedHelpDialogue />
    </ForwardRef(Typography)>
    <ForwardRef(Box)
      sx={
        Object {
          "color": [Function],
          "fontSize": "14px",
          "marginLeft": "auto",
          "textAlign": "right",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexWrap": "nowrap",
          }
        }
      >
        <Memo(ForwardRef(InfoOutlinedIcon))
          fontSize="small"
          sx={
            Object {
              "color": [Function],
            }
          }
        />
         
        <ForwardRef(Typography)
          display="inline"
          sx={
            Object {
              "fontSize": "14px",
              "paddingLeft": "6px",
            }
          }
        >
          searchBox.limited_results_message {maxNumResults:{tabs:{datasetTab:false,datafileTab:false,investigationTab:false},selectAllSetting:true,settingsLoaded:false,sideLayout:false,searchableEntities:[investigation,dataset,datafile],maxNumResults:300}}
        </ForwardRef(Typography)>
      </div>
    </ForwardRef(Box)>
  </div>
</Styled(ForwardRef(Box))>
`;

exports[`SearchBoxContainerSide - Tests renders searchBoxContainerSide correctly 1`] = `
<ForwardRef(Grid)
  alignItems="stretch"
  container={true}
  direction="column"
  id="container-searchbox"
  justifyContent="flex-start"
>
  <ForwardRef(Grid)
    item={true}
  >
    <ForwardRef(Box)
      px={2}
    >
      <SearchTextBox
        initiateSearch={[MockFunction]}
        onChange={[MockFunction]}
        searchText=""
      />
    </ForwardRef(Box)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
  >
    <ForwardRef(Box)
      px={2}
    >
      <Connect(SelectDates)
        initiateSearch={[MockFunction]}
      />
    </ForwardRef(Box)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
  >
    <Connect(CheckboxesGroup) />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
  >
    <ForwardRef(Box)
      mx={5}
      pb={2}
    >
      <TranslatedSearchButton
        initiateSearch={[MockFunction]}
      />
    </ForwardRef(Box)>
  </ForwardRef(Grid)>
</ForwardRef(Grid)>
`;
